hw/ip/spi_device/data/spi_device_testplan.hjson

```diff
@@ -146,5 +165,143 @@
+              followed by the register value.
+            - Compare this value with the expected value.
+
+            TODO: Clarify tpm_mode FIFO mode within the register table.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: tpm_fifo_write
+      desc: '''
+            - Set tpm_mode to 0 in the TPM configuration register.
+            - Set en to 1 in the TPM configuration register.
+            - Randomise other fields in the TPM configuration register.
+
+            TODO: Clarify purpose of TPM_ACCESS_x registers in the spec.
+
+            - Send TPM write command with a randomised address.
+
+            TODO: Clarify the difference between FIFO and CRB mode, can the
+                  command not be written into CMD_ADDR?
+
+            - Based on FIFO status, check SPI bus to confirm WAIT or START sent.
+            - Read TPM_CMD_ADDR then read the write FIFO data when the FIFO is
+              available.
+            - Compare this with the expected value.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: tpm_crb_read
+      desc: '''
+            - Set tpm_mode to 1 in the TPM configuration register.
+            - Set en to 1 in the TPM configuration register.
+            - Randomise other fields in the TPM configuration register.
+
+            TODO: Clarify purpose of TPM_ACCESS_x registers in the spec.
+
+            - Send TPM read command over the SPI bus with a randomised address.
+            - Check TPM_STATUS.cmdaddr_notempty, INTR_STATE.tpm_header_notempty
+              and tpm_header_not_empty interrupt are asserted.
+
+            TODO: Clarify in spec "The SW prepares the register value and
+                  writes the value into the read FIFO." [TPM READ 4]
+
+            - Confirm that the TPM submodule sends WAIT until the read FIFO is
+              available.
+            - When available, confirm that the TPM submodule sends START
+              followed by the register value.
+            - Compare this value with the expected value.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: tpm_crb_write
+      desc: '''
+            - Set tpm_mode to 1 in the TPM configuration register.
+            - Set en to 1 in the TPM configuration register.
+            - Randomise other fields in the TPM configuration register.
+
+            TODO: Clarify purpose of TPM_ACCESS_x registers in the spec.
+
+            - Send TPM write command with a randomised address.
+
+            TODO: Clarify the difference between FIFO and CRB mode, can the
+                  command not be written into CMD_ADDR?
+
+            - Based on FIFO status, check SPI bus to confirm WAIT or START sent.
+            - Read TPM_CMD_ADDR then read the write FIFO data when the FIFO is
+              available.
+            - Compare this with the expected value.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: tpm_max_xfer
```

Comment:
it only supports value 2. designer just updated the default value.
It's just a simple RO register. the automatic CSR tests can cover it.